PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -1133,12 +1132,20 @@ FindImage (
     return EFI_NOT_FOUND;
   }
 
+  // Get the number of payloads
+  PayloadItemCount = CapHeader->PayloadItemCount;
+  if (PayloadItemCount > MAX_FW_COMPONENTS) {
+    DEBUG((DEBUG_ERROR, "# of payloads(%d) in capsule exceeds max(%d). The excess payloads will be ignored\n",
+          PayloadItemCount, MAX_FW_COMPONENTS));
+    PayloadItemCount = MAX_FW_COMPONENTS;
```

Comment:
FindImage does not need to check for MAX_FW_COMPONENTS. 
The MAX_FW_COMPONENTS is defined for the max # of entries (FW_UPDATE_COMP_STATUS) in RSVD. It has a upper bound because of the size of RSVD. 

However, the main goal of FindImage() to find a payload in capsule. A payload may locate after MAX_FW_COMPONENTs, and it should be still searchable. For example, CSMD may locate after MAX_FW_COMPONENTS, but it should be still valid.

Reply:
In ProcessCapsule(), the payloads in a capsule being traversed are limited to the first MAX_FW_COMPONENTS payloads. The payloads beyond are ignored. All the traversed payloads are captured in FwUpdCompStatus[ ]. So there is no way to find a payload located after MAX_FW_COMPONENTS.

Reply:
Incorrect. Imagine CSME + CSMD as an example. CSME is within MAX_FW_COMPONENTS, but CSMD is placed outside of MAX_FW_COMPONENTS. This kind of placement is still valid. Unless you want to disallow this placement.